packages/calcite-components/src/tests/commonTests.ts

```diff
@@ -965,125 +959,178 @@ export async function disabled(
+  };
 
-    component.setProperty("disabled", true);
-    await page.waitForChanges();
+  async function getFocusTarget(page: E2EPage, tag: string, focusTarget: FocusTarget): Promise<string> {
+    return focusTarget === "host" ? tag : await page.evaluate(() => document.activeElement?.tagName.toLowerCase());
+  }
 
-    expect(component.getAttribute("aria-disabled")).toBe("true");
+  const getTabAndClickFocusTarget = async (page: E2EPage, tag: string): Promise<string[]> => {
+    const focusTarget = options.focusTarget;
+    const focusTargetString = await getFocusTarget(page, tag, focusTarget as FocusTarget);
 
-    await page.click(tag);
-    await expectToBeFocused("body");
+    const [tabFocusTarget, clickFocusTarget] =
+      typeof focusTarget === "object" ? [focusTarget.tab, focusTarget.click] : [focusTargetString, focusTargetString];
 
-    await component.callMethod("click");
-    await expectToBeFocused("body");
+    return [tabFocusTarget, clickFocusTarget];
+  };
 
-    assertOnMouseAndPointerEvents(eventSpies, (spy) =>
-      expect(spy).toHaveReceivedEventTimes(eventsExpectedToBubble.includes(spy.eventName) ? 2 : 1)
-    );
+  const getShadowFocusableCenterCoordinates = async (page: E2EPage, tabFocusTarget: string): Promise<number[]> => {
+    return await page.$eval(tabFocusTarget, (element: HTMLElement) => {
+      const focusTarget = element.shadowRoot.activeElement || element;
+      const rect = focusTarget.getBoundingClientRect();
 
-    return;
-  }
-
-  async function getFocusTarget(focusTarget: FocusTarget): Promise<string> {
-    return focusTarget === "host" ? tag : await page.evaluate(() => document.activeElement?.tagName.toLowerCase());
-  }
+      return [rect.x + rect.width / 2, rect.y + rect.height / 2];
+    });
+  };
 
-  await page.keyboard.press("Tab");
+  it("remains non-focusable and disabled while receiving a click event for components that do not allow focusing", async () => {
+    const { page, tag } = await getTagAndPage(componentTestSetup);
 
-  let tabFocusTarget: string;
-  let clickFocusTarget: string;
+    const component = await page.find(tag);
+    await skipAnimations(page);
+    await addClickListenersWithRedirectPrevention(page, tag);
 
-  if (typeof options.focusTarget === "object") {
-    tabFocusTarget = options.focusTarget.tab;
-    clickFocusTarget = options.focusTarget.click;
-  } else {
-    tabFocusTarget = clickFocusTarget = await getFocusTarget(options.focusTarget);
-  }
+    const eventSpies = await createEventSpiesForExpectedEvents(component);
 
-  expect(tabFocusTarget).not.toBe("body");
-  await expectToBeFocused(tabFocusTarget);
+    expect(component.getAttribute("aria-disabled")).toBeNull();
 
-  const [shadowFocusableCenterX, shadowFocusableCenterY] = await page.$eval(tabFocusTarget, (element: HTMLElement) => {
-    const focusTarget = element.shadowRoot.activeElement || element;
-    const rect = focusTarget.getBoundingClientRect();
+    if (options.focusTarget === "none") {
```

Comment:
This test still uses resources to set up the test when the focus target is not `none`. I think this one needs to get merged into `it("prevents focusing via keyboard and mouse")` where you have different focus assertions depending on `options.focusTarget`.

Reply:
Take a look at the merge of those 2 tests now, does the combined test need a new title?

Reply:
Looks great, thanks!